### PR TITLE
Instantiate presets before plugins

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| License                  | MIT

This is only needed for https://github.com/babel/babel/pull/12151, but I'm sending them as two separate PR to make it easier to review them one by one (since the diffs overlap, but are conceptually separate).

The e2e failure makes me think that I got plugins ordering wrong, I'll check.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/11689"></a>

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
Wow, preset ordering is hard. They _sometimes_ run bottom to top, and _sometimes_ top to bottom :upside_down_face: 
In the old code this wasn't explicit, but was handled due to a specific order of push/unshift calls. I have now added an if to switch between push and unshift because I couldn't figure out how to properly reflect the old behavior.

File: packages/babel-core/test/api.js [hunk 1]
```diff
@@ -423,6 +423,8 @@ describe("api", function () {
         "argone;",
         "five;",
         "six;",
+        "twentyone;",
+        "twentytwo;",
```
Comment (nicolo-ribaudo) [inline packages/babel-core/test/api.js:427]: Before the last commit, this was
```js
twentyone
twentytwo
five
six
```

And it was causing a failure in the CRA e2e test.

File: packages/babel-core/src/config/full.js [hunk 2]
```diff
@@ -199,12 +188,32 @@ export default gensync<[any], ResolvedConfig | null>(function* loadFullConfig(
     .map(plugins => ({ plugins }));
   opts.passPerPreset = opts.presets.length > 0;
 
+  if (context.filename?.includes("RuntimeErrorContaine")) {
+    console.log(initialPluginsDescriptors.map(p => p.key));
```
Comment (JLHwung) [inline packages/babel-core/src/config/full.js:192]: 🤔Is that for debug purpose?

**Review by JLHwung — APPROVED** [review]
LGTM other than a nit.

File: packages/babel-core/src/config/full.js (hunk 2 above)
Comment (nicolo-ribaudo) [reply packages/babel-core/src/config/full.js:192]: Yes 🙃
I'll remove it tomorrow

**Review by existentialism — APPROVED** [review]

**Comment by nicolo-ribaudo** [comment]
I'm marking this as "do not merge" until I finish the other PR, so that I'm sure that what I did here is only and everything I need.

**Comment by nicolo-ribaudo** [comment]
Merging into `feat-7.12.0/config-loading-updates`

**Pull request merged**